app/components/data/action/add/file-group-create-form.component.ts

```diff
@@ -0,0 +1,152 @@
+        const observable = this.accountService.currentAccount.cascade((account) => {
+            return this.adalService.accessTokenFor(account.subscription.tenantId, batch).cascade((batchToken) => {
+                return this.adalService.accessTokenFor(account.subscription.tenantId, arm).cascade((armToken) => {
+                    return this.pythonRpcService.call("create_file_group", [
+                        batchToken,
+                        armToken,
+                        formData.name,
+                        formData.includeSubDirectories ? path.join(formData.folder, "**\\*") : formData.folder,
+                        formData.options,
+                        account.toJS(),
+                    ]).catch((error) => {
+                        return Observable.throw(ServerError.fromPython(error));
+                    });
+                });
+            });
+        }).share();
+
+        observable.subscribe({
+            next: (data) => {
+                const message = `${data.uploaded} files were successfully uploaded to the file group`;
+                this.storageService.onFileGroupAdded.next(`${this.storageService.ncjFileGroupPrefix}${formData.name}`);
+                this.notificationService.success("Create file group", message, { persist: true });
+            },
+            error: (error) => {
+                log.error("Failed to create form group", error);
+            },
+        });
+
+        return observable;
+    }
+
+    public dtoToForm(fileGroup: FileGroupCreateDto): CreateFileGroupModel {
+        this.editing = true;
+        this.title = "Add folder to file group";
+        this.description = "Add another folder to an already existing file group";
+        this.form.controls.name.disable();
+
+        return fileGroupToFormModel(fileGroup);
+    }
+
+    public formToDto(data: CreateFileGroupModel): FileGroupCreateDto {
+        return createFileGroupFormToJsonData(data);
+    }
+
+    @autobind()
+    public selectFolder(changeEvent: Event) {
+        const element = changeEvent.srcElement as any;
+        this.form.controls["folder"].markAsTouched();
+
+        if (element.files.length > 0) {
+            this.folder = element.files[0].path;
+            this.form.controls["folder"].setValue(this.folder);
+        } else {
+            this.folder = null;
+            this.form.controls["folder"].setValue(null);
+        }
+    }
+
+    public hasValidFolder(): boolean {
+        return this.folder && this.form.controls["folder"].valid;
+    }
+
+    /**
+     * Async validator to check if a given file-group exists.
+     * If it does exist then we inform the user that they will be modifying
+     * the existing group and not creating a new one.
+     */
+    private _validateFileGroupName(control: FormControl): Promise<any> {
+        return new Promise ((resolve) => {
+            setTimeout(() => {
```

Comment:
because i don't want it to fire on every keypress. google suggested adding a timeout to give the user a chance to type what they want.